applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/formfinding_updated_reference_strategy.hpp

```diff
@@ -869,185 +338,50 @@ namespace Kratos
-            {
-                std::cout << "Solution obtained = " << Dx << std::endl;
-                std::cout << "RHS  = " << b << std::endl;
-            }
-            else if (this->GetEchoLevel() == 3) //if it is needed to print the debug info
-            {
-                //             std::cout << "After first system solution" << std::endl;
-                std::cout << "SystemMatrix = " << A << std::endl;
-                std::cout << "Solution obtained = " << Dx << std::endl;
-                std::cout << "RHS  = " << b << std::endl;
-            }
-            else if (this->GetEchoLevel() == 4) //print to matrix market file
-            {
-                std::stringstream matrix_market_name;
-                matrix_market_name << "A_" << BaseType::GetModelPart().GetProcessInfo()[TIME] << "_" << iteration_number << ".mm";
-                TSparseSpace::WriteMatrixMarketMatrix((char*)(matrix_market_name.str()).c_str(), A, false);
-
-                std::stringstream matrix_market_vectname;
-                matrix_market_vectname << "b_" << BaseType::GetModelPart().GetProcessInfo()[TIME] << "_" << iteration_number << ".mm.rhs";
-                TSparseSpace::WriteMatrixMarketVector((char*)(matrix_market_vectname.str()).c_str(), b);
-            }
-        }
-
-
-        //**********************************************************************
-        //**********************************************************************
+        };
 
-        void MaxIterationsExceeded()
+        void EchoInfo(const unsigned int IterationNumber) override
         {
-            if (this->GetEchoLevel() != 0 && BaseType::GetModelPart().GetCommunicator().MyPID() == 0)
+            BaseType::EchoInfo(IterationNumber);
+
+            if (mPrintIterations)
             {
-                std::cout << "***************************************************" << std::endl;
-                std::cout << "******* ATTENTION: max iterations exceeded ********" << std::endl;
-                std::cout << "***************************************************" << std::endl;
+                KRATOS_ERROR_IF_NOT(mpIterationIO) << " IterationIO is uninitialized!" << std::endl;
+                mpIterationIO->WriteNodalResults(DISPLACEMENT, BaseType::GetModelPart().Nodes(), IterationNumber, 0);
             }
-
         }
 
-        /**
-        * function to perform expensive checks.
-        * It is designed to be called ONCE to verify that the input is correct.
-        */
-
-        int Check() override
+        void InitializeIterationIO()
         {
-            KRATOS_TRY
-
-                BaseType::Check();
-
-            GetBuilderAndSolver()->Check(BaseType::GetModelPart());
-
-            GetScheme()->Check(BaseType::GetModelPart());
-
-            mpConvergenceCriteria->Check(BaseType::GetModelPart());
-
-            return 0;
-
-            KRATOS_CATCH("")
+            mpIterationIO = Kratos::make_shared<IterationIOType>(
```

Comment:
can you replace with make_unique please?

Reply:
done

Reply:
that doesn't work for me. And I did not find anything else like this in Kratos.